number guessing game/extension_functions.py

```diff
@@ -2,7 +2,77 @@
 Game extension functions. These functions add extra functionalities to the game.
 """
 
-from game_constants import SCORES_PER_DIFFICULTY, BONUS_MULTIPLIERS
+import json
+from os import path
+from game_constants import SCORES_PER_DIFFICULTY, BONUS_MULTIPLIERS, HIGH_SCORES_FILE
+
+
+def load_high_scores(file_name: str) -> dict:
+    """
+    Load game high score(s) from a Json file or create an emtpy file
+    if file doesn't exist.
+
+    :param file_name: Name of the file
+    :return: A dict containing the high score(s)
+    """
+
+    if not path.exists(file_name):
+        # create the file
+        with open(file_name, 'w', encoding='utf-8') as file_stream_obj:
+            json.dump({}, file_stream_obj)
+
+    # file exist
+    with open(file_name, 'r', encoding='utf-8') as file_obj:
+        return json.load(file_obj)
+
+
+def update_high_scores(player_name: str, score: int):
```

Comment:
This works fine. I think you can extend this further by adding additional functionalities or maybe another function. Something that will also be able to take into account the game levels . Each level depit different score.

Reply:
Okay, thanks for your feedback.  I will look into that, adding such a feature will help improve the overall user's experience of the game. Thanks for your wonderful feedback.